### PR TITLE
feature/oauth

### PR DESCRIPTION
**Pull request by KunLee76**
- [x]  Use Middleware for authorization
- [x]  Consumer retrieve token
- [x]  Replace the service name in each HTTP request of the producer with the OpenAPI name
- [ ]  Complete unit test of router_auth_check_test.go
- [x]  Complete code checking with Golangci-lint and modifying code with GOFMT

### CODE REVIEW

File: internal/sbi/consumer/nf_management.go [hunk 1]
```diff
@@ -102,7 +109,7 @@ func SendRegisterNFInstance(nrfUri, nfInstanceId string, profile models.NfProfil
 func SendDeregisterNFInstance() (problemDetails *models.ProblemDetails, err error) {
 	logger.ConsumerLog.Infof("Send Deregister NFInstance")
 
-	ctx, pd, err := pcf_context.GetSelf().GetTokenCtx("nnrf-nfm", "NRF")
+	ctx, pd, err := pcf_context.GetSelf().GetTokenCtx("nnrf-nfm", "models.NfType_NRF")
```
Comment (ianchen0119) [inline internal/sbi/consumer/nf_management.go:112]: Should be:
```
ctx, pd, err := pcf_context.GetSelf().GetTokenCtx("nnrf-nfm", models.NfType_NRF)
```

File: internal/sbi/consumer/nf_discovery.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func SendSearchNFInstances(
 	configuration.SetBasePath(nrfUri)
 	client := Nnrf_NFDiscovery.NewAPIClient(configuration)
 
-	ctx, _, err := pcf_context.GetSelf().GetTokenCtx("nnrf-disc", "NRF")
+	ctx, _, err := pcf_context.GetSelf().GetTokenCtx("nnrf-nfm", models.NfType_NRF)
```
Comment (ianchen0119) [inline internal/sbi/consumer/nf_discovery.go:25]: Should be:
```
ctx, _, err := pcf_context.GetSelf().GetTokenCtx("nnrf-disc", models.NfType_NRF)
```

File: internal/sbi/consumer/influenceDataSubscription.go [hunk 3]
```diff
@@ -70,9 +73,13 @@ func RemoveInfluenceDataSubscription(ue *pcf_context.UeContext, subscriptionID s
 		logger.ConsumerLog.Warnf("Can't find corresponding UDR with UE[%s]", ue.Supi)
 		return &problemDetail, nil
 	}
+	ctx, pd, err := pcf_context.GetSelf().GetTokenCtx("nnrf-nfm", models.NfType_NRF)
+	if err != nil {
+		return pd, err
+	}
 	udrClient := util.GetNudrClient(ue.UdrUri)
 	httpResp, localErr := udrClient.IndividualInfluenceDataSubscriptionDocumentApi.
```
Comment (ianchen0119) [inline internal/sbi/consumer/influenceDataSubscription.go:81]: This client is used to communicate with UDR.
The context should be generated by UDR's service and UDR nfType.

**Review by ianchen0119 — CHANGES_REQUESTED** [review]
Please check All of the use of the `GetTokenCtx(scope string, targetNF models.NfType)`, the scope and targetNF are not always equal to `nrf-nfm` and `NRF`.

File: internal/sbi/producer/smpolicy.go [hunk 4]
```diff
@@ -319,7 +320,7 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
 	bsfUri := consumer.SendNFInstancesBSF(pcf_context.GetSelf().NrfUri)
 	if bsfUri != "" {
 		bsfClient := util.GetNbsfClient(bsfUri)
-		_, resp, err = bsfClient.PCFBindingsCollectionApi.CreatePCFBinding(ctx, pcfBinding)
+		_, resp, err = bsfClient.PCFBindingsCollectionApi.CreatePCFBinding(context.Background(), pcfBinding)
```
Comment (ianchen0119) [inline internal/sbi/producer/smpolicy.go]: @KunLee76 @andy89923 
I think it also needs the token context.

Comment (KunLee76) [reply internal/sbi/producer/smpolicy.go]: Thank you for all your assistance

**Review by andy89923 — COMMENTED** [review]
LGTM!

**Review by andy89923 — APPROVED** [review]

**Review by ianchen0119 — APPROVED** [review]

**Pull request merged**